--- security/manager/ssl/src/nsPKCS12Blob.cpp.orig	2004-11-28 19:08:30.000000000 +0100
+++ security/manager/ssl/src/nsPKCS12Blob.cpp	2004-11-28 20:33:19.387772206 +0100
@@ -187,7 +187,7 @@
   mToken->GetTokenName(getter_Copies(tokenName));
   {
     NS_ConvertUTF16toUTF8 tokenNameCString(tokenName);
-    slot = PK11_FindSlotByName(tokenNameCString.get());
+    slot = PK11_FindSlotByName(NS_CONST_CAST(char*,tokenNameCString.get()));
   }
   if (!slot) {
     srv = SECFailure;
